openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -141,6 +153,8 @@ def get_content_tags(
     )
 
 
+# FixMe: The following method (tag_content_object) is only used in tasks.py for auto-tagging. To tag object we are
+# using oel_tagging.tag_object and checking permissions via rule overrides.
 def tag_content_object(
```

Comment:
I flagged it to see if we need to do something about it.

Reply:
I don't really understand the issue here. Is this method not needed? Or should we actually be using it instead of oel_tagging.tag_object?

Reply:
This method is unnecessary, but the auto-tagging task uses it instead of the `oel_tagging.tag_object`. Should we fix this as part of this task, or will we handle it in a follow-up task?

Reply:
If it's an easy fix feel free to just fix it now :)

Reply:
Hm.. This method also uses the ContentObjectTag proxy model. Using `oel_tagging.tag_object` would skip that. I think we are not doing a lot with ContentObjectTag now.. Should we also remove the model?
What do you think @pomegranited?

Reply:
@rpenido I believe ContentObjectTag validates that the ID is an opaque key, which is useful.

Reply:
https://github.com/openedx/edx-platform/blob/d4536ed79d771aa502da45d125f7ede0fe5d9b90/openedx/core/djangoapps/content_tagging/rules.py#L217-L231

Reply:
OK, we might not need it at all then.

Reply:
I think this fix/change deserves a separate PR.

Reply:
Here: https://github.com/openedx/edx-platform/pull/34146
It's not done yet (I'll comment in the PR).

@pomegranited Could you check if it is what you expected?